lecture_1/hw/math_plain_asgi.py

```diff
@@ -1,8 +1,17 @@
-from typing import Any, Awaitable, Callable
+from lecture_1.hw.Factorial import factorial
+from lecture_1.hw.fibonacci import FibonacciHandler
+from lecture_1.hw.mean import Mean
+from lecture_1.hw.base import send_response
 
 
-async def app(
-    scope: dict[str, Any],
-    receive: Callable[[], Awaitable[dict[str, Any]]],
-    send: Callable[[dict[str, Any]], Awaitable[None]],
-) -> None: ...
+async def app(scope, receive, send):
+    path = scope["path"]
+
+    if path == "/factorial":
+        await factorial(scope, receive, send).handle()
+    elif "/fibonacci" in path:
+        await FibonacciHandler(scope, receive, send).handle()
+    elif "/mean" in path:
+        await Mean(scope, receive, send).handle()
```

Comment:
```suggestion
    if path == "/factorial":
        await FactorialHandler(scope, receive, send).handle()
    elif "/fibonacci" in path:
        await FibonacciHandler(scope, receive, send).handle()
    elif "/mean" in path:
        await MeanHandler(scope, receive, send).handle()
```
а еще лучше в классе Base сделать метод match, который возвращает true или false

for e in endpoints:
if e.match(path): ...

Почитай про паттерн "стратегия"